### PR TITLE
Add variable for customer gateway device name

### LINKED ISSUE #56: Make Customer Gateway Device Name optional
### Describe the Bug

Since 1.3.1 the customer gateway name is defaulted to the `module.this.id`. This is fine if building a new VPN but upgrading an exciting VPN causes a CGW rebuild and potential short downtime to a VPN connection. AWS have the device name as an optional field and this should be reflected.

### Expected Behavior

The ability to null this value

### Steps to Reproduce

Create a VPN pre 1.3.1 and upgrade

### Screenshots

_No response_

### Environment

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by oc-christopher-billett**
## what

Set the customer gateway device name to be a variable

## why

To stop it rebuilding on existing VPN's with it not set

## references

closes #56

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -21,7 +21,7 @@ resource "aws_vpn_gateway" "default" {
 # https://www.terraform.io/docs/providers/aws/r/customer_gateway.html
 resource "aws_customer_gateway" "default" {
   count            = local.enabled && var.customer_gateway_ip_address != null ? 1 : 0
-  device_name      = module.this.id
+  device_name      = var.customer_gateway_device_name
```
Comment (jamengual) [inline main.tf:24]: this is not backwards compatible.
please add a tenary to always default to module.this.id if var.customer_gateway_device_name is not defined.

**Review by jamengual — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (oc-christopher-billett) [reply main.tf]: @jamengual - I have added the the default to be an empty string which will be converted to the module id. But to allow backwards compatibility null is a valid option

**Comment by oc-christopher-billett** [comment]
@johncblandii @jamengual - Could I get a review on this again please?

**Review by jamengual — APPROVED** [review]

**Comment by jamengual** [comment]
Thanks @oc-christopher-billett for the contribution

File: main.tf [hunk 2]
```diff
@@ -21,7 +21,7 @@ resource "aws_vpn_gateway" "default" {
 # https://www.terraform.io/docs/providers/aws/r/customer_gateway.html
 resource "aws_customer_gateway" "default" {
   count            = local.enabled && var.customer_gateway_ip_address != null ? 1 : 0
-  device_name      = module.this.id
+  device_name      = var.customer_gateway_device_name == "" ? module.this.id : var.customer_gateway_device_name
```
Comment (johncblandii) [inline main.tf:24]: Any reason to not use `null` for the var?

Comment (jamengual) [reply main.tf:24]: mmm true, it could have been nulll.

**Pull request merged**